core/src/main/java/org/apache/iceberg/AllEntriesTable.java

```diff
@@ -64,9 +84,18 @@ protected TableScan newRefinedScan(Table table, Schema schema, TableScanContext
 
     @Override
     protected CloseableIterable<FileScanTask> doPlanFiles() {
-      CloseableIterable<ManifestFile> manifests =
-          reachableManifests(snapshot -> snapshot.allManifests(table().io()));
-      return BaseEntriesTable.planFiles(table(), manifests, tableSchema(), schema(), context());
+      return new ParallelIterable<>(
```

Comment:
Looks like we don't close the ParallelIterable like in the original reachableManifests method. 

Also looks like we are going to call planFiles many more times than the original.

My first thought is to keep the logic but use a Pair<ManifestFile, Snapshot> where ManifestFile is used now.  How about adapting the correct reachableManifest code into a new more generic method like

T traverse(Function<Snapshot, T> func)

and have reachableManifests use that?

Reply:
@szehon-ho Thanks for your feedback!

> Looks like we don't close the ParallelIterable like in the original reachableManifests method.

Good catch! Let me fix it.

> Also looks like we are going to call planFiles many more times than the original.
Let me think about it, thanks.

Reply:
> Looks like we don't close the ParallelIterable like in the original reachableManifests method.

Good catch! Let me fix it.

> Also looks like we are going to call planFiles many more times than the original.

Let me think about it, thanks.